src/vcpkg/install.cpp

```diff
@@ -621,11 +621,53 @@ namespace vcpkg::Install
         }
     }
 
-    CMakeUsageInfo get_cmake_usage(const Filesystem& fs, const InstalledPaths& installed, const BinaryParagraph& bpgh)
+    static constexpr StringLiteral ADD_LIBRARY_CALL = "add_library(";
+
+    static const char* find_add_library(const char* real_first, const char* first, const char* last)
+    {
+        for (;;)
+        {
+            first = std::search(first, last, ADD_LIBRARY_CALL.begin(), ADD_LIBRARY_CALL.end());
+            if (first == real_first || first == last || !Parse::ParserBase::is_word_char(*(first - 1)))
```

Comment:
```suggestion
            if (first == real_first || first == last || !Parse::ParserBase::is_word_char(first[-1]))
```

Reply:
I know this is equivalent, but I _really_ don't like array indexing a pointer; I'd prefer to treat it as an un-indexable iterator

Reply:
> I'd prefer to treat it as an un-indexable iterator

Note that random-access iterators require `op[]`: http://eel.is/c++draft/random.access.iterators

<img width="545" alt="image" src="https://user-images.githubusercontent.com/1544943/159346892-9beac99f-cb61-460f-aa4e-57efff62870e.png">

No additional change requested; just wanted to note this.